daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/Assert.scala

```diff
@@ -128,7 +128,7 @@ object Assert extends Assert {
   //
 
   def usageError(message: String = "Usage error."): Nothing = {
-    abort(message)
+    toss(new IllegalStateException(message + "\n" + shortBacktrace))
```

Comment:
Do we need to include shortBacktrace? I think the `IllegalStateException` should contain a stack trace that API users can output if they want.

Reply:
I think we want to revert this change? The `usage` functions are now about internal API checks that indicate bugs in Daffoil, so an abort is probably the right thing to do.